rpcserver.go

```diff
@@ -6748,6 +6752,35 @@ func (r *rpcServer) ListPermissions(_ context.Context,
 	}, nil
 }
 
+// CheckMacaroonPermissions checks the caveats and permissions of a macaroon.
+func (r *rpcServer) CheckMacaroonPermissions(ctx context.Context,
+	req *lnrpc.CheckMacPermRequest) (*lnrpc.CheckMacPermResponse, error) {
+
+	// Turn grpc macaroon permission into bakery.Op for the server to
+	// process.
+	permissions := make([]bakery.Op, len(req.Permissions))
```

Comment:
This is a common gotcha in golang... If you use pre-allocation, then you either need to set the size to 0 and only the pre-allocate the capacity (`make([]bakery.Op, 0, len(req.Permissions))`) or you need to use the index to assign values instead of `append()`:

```
	for idx, perm := range req.Permissions {
		permissions[idx] = bakery.Op{
			Entity: perm.Entity,
			Action: perm.Action,
		}
	}
```

Otherwise you'll get a slice that is double the length of what you expect with the first half all being nil.